app/domain/authentication/authn_oidc/authenticate.rb

```diff
@@ -51,6 +52,14 @@ def new_token(input)
           username: input.username
         )
       end
+
+      def conjur_username(request_body)
+        id_token_username_field = "preferred_username"
+        conjur_username = request_body.id_token[id_token_username_field]
+        raise IdTokenFieldNotFound, id_token_username_field unless conjur_username
```

Comment:
Check if there is any possibility that the user name will be empty from some reason

Reply:
good call. I handled this and added a UT.

btw, the expected behaviour IMO is the same as with no preferred_username at all.